cmdstanpy/model.py

```diff
@@ -474,9 +474,14 @@ def compile(
                     self._compiler_options.add(compiler_options)
         exe_target = os.path.splitext(self._stan_file)[0] + EXTENSION
         if os.path.exists(exe_target):
-            src_time = os.path.getmtime(self._stan_file)
             exe_time = os.path.getmtime(exe_target)
-            if exe_time > src_time and not force:
+            included_files = [self._stan_file]
+            included_files.extend(self.src_info().get('included_files', []))
+            out_of_date = any(
+                os.path.getmtime(included_file) > exe_time
+                for included_file in included_files
```

Comment:
~Even after using `src_info`, this still needs to take `include_paths` into account, no?~

Oh, actually, I just checked and `src_info` returns the actual included file paths, not just the relative paths. That's much better.

Regardless, I think it is still worth adding a test for an include which uses `include_paths` set to something non-default.